Unauthorized response in GetPackageRepositorySummaries

```
Invalid GetPackageRepositorySummaries response from the plugin helm.packages: rpc error: code = Unauthenticated desc = Authorization required to get the AppRepository 'all' due to 'Unauthorized'
```

We're trying to upgrade to kubeapps 2.6.3 (latest release) and I've encountered a strange bug when trying to load the `Catalog`.

Seemingly due to an Unauthorized GRPC response, the dashboard redirects back to `/` and I end up right where I started (i.e. main dashboard page viewing applications).

1. Load kubeapps
2. Click `Catalog`
3. Redirected back to dashboard

![image](https://user-images.githubusercontent.com/2043821/219078380-4b01772d-5489-4bc4-91bc-ed3a109d83c2.png)
![image](https://user-images.githubusercontent.com/2043821/219078427-dbd75685-01f8-4707-9082-f8e09d0d7dac.png)
![image](https://user-images.githubusercontent.com/2043821/219078489-53f8a556-771e-4100-994f-4dc72187d671.png)

Is there some new configuration now that we need to do to permit our users to view package repositories?

We have 2 `AppRepository` configured in this cluster (BTW what is the difference to the "PackageRepository" concept?):

```
(⎈ default:kubeapps)mcampbell-1 :: coreweave/k8s-services/dev ‹master*› » k neat get apprepository
apiVersion: v1
items:
- apiVersion: kubeapps.com/v1alpha1
  kind: AppRepository
  metadata:
    annotations:
      meta.helm.sh/release-name: kubeapps
      meta.helm.sh/release-namespace: kubeapps
    labels:
      app.kubernetes.io/instance: kubeapps
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: kubeapps
      helm.sh/chart: kubeapps-12.2.1-dev0
    name: ant-repo
    namespace: kubeapps
  spec:
    syncJobPodTemplate:
      spec:
        securityContext:
          runAsUser: 1001
    type: helm
    url: http://charts.tenant-rizzo.ord1.ingress.coreweave.cloud/
- apiVersion: kubeapps.com/v1alpha1
  kind: AppRepository
  metadata:
    annotations:
      meta.helm.sh/release-name: kubeapps
      meta.helm.sh/release-namespace: kubeapps
    labels:
      app.kubernetes.io/instance: kubeapps
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: kubeapps
      helm.sh/chart: kubeapps-12.2.1-dev0
    name: coreweave
    namespace: kubeapps
  spec:
    syncJobPodTemplate:
      spec:
        securityContext:
          runAsUser: 1001
    type: helm
    url: http://helm.corp.ingress.ord1.coreweave.com/
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

I noticed that the request is using context of `cluster="" namespace=""`.

I hard-coded those values to `cluster="default" namespace="kubeapps"` and these requests started working as expected.

```git
diff --git a/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories.go b/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories.go
index 5782e3907..004927670 100644
--- a/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories.go
+++ b/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories.go
@@ -563,12 +563,12 @@ func (s *Server) repoSummaries(ctx context.Context, cluster string, namespace st

 // GetPkgRepositories returns the list of package repositories for the given cluster and namespace
 func (s *Server) GetPkgRepositories(ctx context.Context, cluster, namespace string) ([]*apprepov1alpha1.AppRepository, error) {
-       resource, err := s.getPkgRepositoryResource(ctx, cluster, namespace)
+       resource, err := s.getPkgRepositoryResource(ctx, "default", "kubeapps")
```

```go
// GetPkgRepositories returns the list of package repositories for the given cluster and namespace
func (s *Server) GetPkgRepositories(ctx context.Context, cluster, namespace string) ([]*apprepov1alpha1.AppRepository, error) {
	//resource, err := s.getPkgRepositoryResource(ctx, cluster, namespace)
	resource, err := s.getPkgRepositoryResource(ctx, "default", "kubeapps")
	if err != nil {
		log.Infof("+[ERROR] helm GetPkgRepositories [%v]", err)
		return nil, err
	}
	log.InfoS("+[MATT] s.getPkgRepositoryResource", "resource", resource, "ctx", ctx, "cluster", "default", "namespace", "kubeapps")

	unstructured, err := resource.List(ctx, metav1.ListOptions{})
	log.InfoS("+[MATT] GetPkgRepositories unstructured", "unstructured", unstructured, "err", err)
```
Do you know why the frontend would be sending a gRPC request with that context `cluster="" namespace=""`?

Upon hardcoding I can view the catalog as expected, but cannot actually deploy a chart:

![image](https://user-images.githubusercontent.com/2043821/219471327-666523b4-95e0-4a44-9df7-c832ee0142b1.png)
![image](https://user-images.githubusercontent.com/2043821/219471347-ae23a1b2-1740-416a-b158-81b718659244.png)

--- a/pkg/kube/cluster_config.go
+++ b/pkg/kube/cluster_config.go
@@ -137,7 +137,10 @@ func NewClusterConfig(inClusterConfig *rest.Config, userToken string, cluster st
 		return config, nil
 	}
 
-	if cluster == clustersConfig.KubeappsClusterName {
+	// We cannot assume that if the cluster is the kubeapps cluster that we simply return
+	// the incluster config, because some users set proxies in front of their clusters in
+	// which case the incluster kubernetes.default will skip the proxy.
+	if cluster == clustersConfig.KubeappsClusterName && clusterConfig.APIServiceURL == "" {
 		return config, nil
 	}
 

--- a/pkg/kube/cluster_config_test.go
+++ b/pkg/kube/cluster_config_test.go
@@ -45,6 +45,39 @@ func TestNewClusterConfig(t *testing.T) {
 			},
 		},
 		{
+			name:      "returns a cluster config with explicit apiServiceURL and cert even for the kubeapps default cluster, when specified",
+			userToken: "token-1",
+			cluster:   "default",
+			clustersConfig: ClustersConfig{
+				KubeappsClusterName: "default",
+				Clusters: map[string]ClusterConfig{
+					"default": {
+						APIServiceURL:                   "https://proxy.example.com:7890",
+						CertificateAuthorityData:        "Y2EtZmlsZS1kYXRhCg==",
+						CertificateAuthorityDataDecoded: "ca-file-data",
+						CAFile:                          "/tmp/ca-file-data",
+					},
+				},
+			},
+			inClusterConfig: &rest.Config{
+				Host:            "https://something-else.example.com:6443",
+				BearerToken:     "something-else",
+				BearerTokenFile: "/foo/bar",
+				TLSClientConfig: rest.TLSClientConfig{
+					CAFile: "/var/run/whatever/ca.crt",
+				},
+			},
+			expectedConfig: &rest.Config{
+				Host:            "https://proxy.example.com:7890",
+				BearerToken:     "token-1",
+				BearerTokenFile: "",
+				TLSClientConfig: rest.TLSClientConfig{
+					CAData: []byte("ca-file-data"),
+					CAFile: "/tmp/ca-file-data",
+				},
+			},
+		},
+		{
 			name:      "returns an in-cluster config when the global packaging cluster token is specified",
 			userToken: "token-1",
 			cluster:   KUBEAPPS_GLOBAL_PACKAGING_CLUSTER_TOKEN,

